model/QtiItemCompiler.php

```diff
@@ -57,25 +57,36 @@ class QtiItemCompiler extends taoItems_models_classes_ItemCompiler implements Se
      * @var string
      */
     const INSTANCE_ITEMRUNNER = 'http://www.tao.lu/Ontologies/TAOItem.rdf#ServiceQtiItemRunner';
-    
+
+    /**
+     * {@inheritDoc}
+     * @see taoItems_models_classes_ItemCompiler::compile()
+     */
+    public function compile()
+    {
+        $report = $this->internalCompile();
+        if ($report->getType() == common_report_Report::TYPE_SUCCESS) {
+            // replace instances with service
+            list($item, $publicDirectory, $privateDirectory) = $report->getData();
+            $report->setData($this->createQtiService($item, $publicDirectory, $privateDirectory));
+        }
+        return $report;
+    }
+
     /**
      * Compile qti item
      *
      * @throws taoItems_models_classes_CompilationFailedException
      * @return tao_models_classes_service_ServiceCall
      */
-    public function compile()
+    protected function internalCompile()
     {
-
+        $item = $this->getResource();
         $publicDirectory = $this->spawnPublicDirectory();
         $privateDirectory = $this->spawnPrivateDirectory();
-        $item = $this->getResource();
 
         $report = new common_report_Report(common_report_Report::TYPE_SUCCESS, __('Published %s', $item->getLabel()));
-        if (!taoItems_models_classes_ItemsService::singleton()->isItemModelDefined($item)) {
```

Comment:
what happened with this check? it should be at least in the compile() no?

Reply:
Afaict this check is not reachable